### PR TITLE
Add Painless Docs for Datetime Inputs

### PR DESCRIPTION
**Pull request by jdconrad**
This changes add documentation for accessing datetimes in Painless scripts from the three most common inputs of params, _source, and doc.

### CODE REVIEW

File: docs/painless/painless-guide/painless-datetime.asciidoc [hunk 1]
```diff
@@ -334,3 +334,230 @@ if (zdt1.isAfter(zdt2)) {
...
+        ...
+    }
+}
+----
++
+** Script:
++
+[source,Painless]
+----
+// check for the existence of fields within the current document
```
Comment (rjernst) [inline docs/painless/painless-guide/painless-datetime.asciidoc]: The contains key check is for existence of a mapping. This is relevant when doing a multi index search. If hte mappinng exists, doc will have the field, it just might not have any values.

**Review by rjernst — APPROVED** [review]
LGTM just a few minor comments.

File: docs/painless/painless-guide/painless-datetime.asciidoc [hunk 2]
```diff
@@ -334,3 +334,230 @@ if (zdt1.isAfter(zdt2)) {
...
+----
+Note the use of a custom DateTimeFormatter.
+
+===== Datetime Input From a Source Document
+
+Use an original {ref}/mapping-source-field.html[source] document as a script
+input to access a numeric datetime or string datetime for a specific field
+within that document. Access to an original source document within a script is
+dependent on the Painless context and is not always available. An original
+source document is most commonly accessible through an input called `_source`
```
[the remark this answers is not in the document]
Comment (jdconrad) [reply docs/painless/painless-guide/painless-datetime.asciidoc]: Good catch.  Added notes.

File: docs/painless/painless-guide/painless-datetime.asciidoc [hunk 3]
```diff
@@ -334,3 +334,230 @@ if (zdt1.isAfter(zdt2)) {
...
+    "input_datetime": 434931327000
+    ...
+}
+----
++
+** Script:
++
+[source,Painless]
+----
+long inputDatetime = _source['input_datetime'];
```
[the remark this answers is not in the document]
Comment (jdconrad) [reply docs/painless/painless-guide/painless-datetime.asciidoc]: Added notes.

File: docs/painless/painless-guide/painless-datetime.asciidoc (hunk 1 above)
Comment (jdconrad) [reply docs/painless/painless-guide/painless-datetime.asciidoc]: I understand. I'll try to reword this to make it clear what is happening.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review!  I've addressed your comments and will commit as soon as CI passes.
